backend/courses/models.py

```diff
@@ -1419,3 +1420,69 @@ def create_or_update_user_profile(sender, instance, created, **kwargs):
     if created and instance.email != "":
         instance.profile.email = instance.email
     instance.profile.save()
+
+
+class Friendship(models.Model):
+    """
+    Used to track friendships along with requests status
+    """
+
+    sender = models.ForeignKey(
+        get_user_model(),
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="sent_friendships",
+        help_text="The person (user) who sent the request.",
+    )
+
+    recipient = models.ForeignKey(
+        get_user_model(),
+        related_name="received_friendships",
+        on_delete=models.CASCADE,
+        null=True,
+        help_text="The person (user) who recieved the request.",
+    )
+
+    class FriendshipStatus(models.TextChoices):
+        SENT = "S", "Sent"
+        ACCEPTED = "A", "Accepted"
+        REJECTED = "R", "Rejected"
+
+    status = models.CharField(
+        max_length=1,
+        choices=FriendshipStatus.choices,
+        default=FriendshipStatus.SENT,
+    )
+
+    def check_friendship(self, user1_id, user2_id):
+        """
+        Checks if two users are friends (lookup by user id)
+        """
+        user1 = User.objects.get(id=user1_id)
+        user2 = User.objects.get(id=user2_id)
+        return (Friendship.objects.filter(sender=user1, recipient=user2, status="A").exists() or 
+                Friendship.objects.filter(sender=user2, recipient=user1, status="A").exists())
+
+    def save(self, *args, **kwargs):
+        if (self.status == self.FriendshipStatus.ACCEPTED):
+            self.accepted_at = timezone.now()
+        if (self.status == self.FriendshipStatus.REJECTED):
+            # any other logic that needs to happen when a friendship is rejected 
+            pass
```

Comment:
I added some logic to reset the "sent_at" and "accepted_at" values to None when the request is rejected (in order to properly set the times in case the request has been sent again.

Reply:
Nice